### PR TITLE
[cxx-interop] Add support for C++ shift operators.

### PR DESCRIPTION
**Pull request by zoecarver**
Support imported C++ `<<` and `>>` operators in Swift.

Refs #32332.

### CODE REVIEW

**Comment by MForster** [comment]
@swift-ci Please test

File: test/Interop/Cxx/operators/non-member-inline.swift [hunk 1]
```diff
@@ -43,4 +43,22 @@ OperatorsTestSuite.test("slash") {
   expectEqual(1, result.value)
 }
 
+OperatorsTestSuite.test("less less (<<)") {
```
Comment (MForster) [inline test/Interop/Cxx/operators/non-member-inline.swift:46]: Let's be consistent and either drop the parentheses in the test names or add them above as well.

**Review by MForster — APPROVED** [review]
Thanks!

**Review by gribozavr — APPROVED** [review]

File: test/Interop/Cxx/operators/non-member-inline.swift (hunk 1 above)
Comment (MForster) [reply test/Interop/Cxx/operators/non-member-inline.swift:46]: I just made this change myself.

**Comment by MForster** [comment]
@swift-ci Please test

**Comment by MForster** [comment]
@swift-ci Please test

**Comment by MForster** [comment]
@swift-ci Please test

File: test/Interop/Cxx/operators/non-member-inline.swift (hunk 1 above)
Comment (zoecarver) [reply test/Interop/Cxx/operators/non-member-inline.swift:46]: Thank you :)

**Pull request merged**